lms/djangoapps/courseware/courses.py

```diff
@@ -121,9 +122,18 @@ def check_course_access(course, user, action, check_if_enrolled=False):
 
     access_response = has_access(user, action, course, course.id)
     if not access_response:
-        # Deliberately return a non-specific error message to avoid
-        # leaking info about access control settings
-        raise CoursewareAccessException(access_response)
+        params = None
+        if isinstance(access_response, StartDateError):
+            start_date = strftime_localized(course.start, 'SHORT_DATE')
+            params = QueryDict(mutable=True)
+            params['notlive'] = start_date
+
+        # If messaging is added for general access problems, make it vague to
+        # not leak info about access control settings.
+        raise CourseAccessRedirect('{dashboard_url}{params}'.format(
```

Comment:
@marcotuts @andy-armstrong: adding a redirect for all access errors breaks a lot of new tests. I'd like to revert that change on this PR and can copy the list of broken tests to a future story that we can choose to do or skip. Otherwise, this is scope creep on this story.